src/Select/Select.tsx

```diff
@@ -82,34 +102,29 @@ export const Select = ({
       {...rest}
       py={py}
       open={isOpen}
-      test={test}
       placement={DropdownPlacement.BOTTOM_START}
-      useTriggerWidth={useTriggerComponentWidth}
-      renderTrigger={({ ref }) => getTriggerComponent(
-        { ...triggerComponentProps },
-        ref as Ref<HTMLInputElement>,
-        triggerComponent
+      useTriggerWidth={useInputComponentWidth}
+      TriggerComponent={() => (
+        InputComponent
+          ? <InputComponent {...inputComponentProps}/>
+          : <SelectInput {...inputComponentProps}/>
       )}
       onOutsideClick={
         () => onOutsideClick
           ? onOutsideClick(isOpen, close)
-          : setIsOpen(false)
+          : close()
       }>
-        <OptionsWrapper fullWidth={useTriggerComponentWidth}>
+        <OptionsWrapper fullWidth={useInputComponentWidth}>
           {children
-            ? children({
-              close: () => setIsOpen(false),
-              value,
-              data
-            })
+            ? children
             : (
-              <SelectList
-                value={value.key ? value : undefined}
-                onChange={(v: SelectListData) => {
+              <SelectList<V extends SelectListData ? V : never>
+                value={value as V extends SelectListData ? V : never}
```

Comment:
This is debatable but I would argue that it's best to just go into `SelectList` and remove the `value` prop. It makes things more complex and bloated than they need to be, `value` is not used in the default item, and lastly, `value` is just the selected item and if we needed it in a `SelectListItem` then we have to make a custom item anyway and we could easily pass the selected item with context for instance. In my opinion `value` is not needed.

Reply:
for me, it brings uncertainty. When you passing the `value`, the state for whole `<Select>` component (input + options) is controlled within single control point.
Otherwise 2 different local states are being used (1 for input and another 1 for options)

Reply:
Normally the options should not be controlled by the select. The only reason they are now is because we decided that we wanted a default implementation for the select. If you made your own you would have to manage the state for the options yourself anyway. 

What I mean is that the default implementation does not need the value, it simply doesn't use it. If you want to make use of the value that is passed you would have to make another implementation so you would have to manage that state outside of the select anyway. 

In that case I would say either remove it completely (there are other ways to achieve the same), or make the value optional and don't pass it to the options component in the default select implementation because it's redundant.

Reply:
Correct, if I got wrong, but seems like the value is actually being used
https://github.com/exivity/ui/blob/master/src/SelectList/SelectList.tsx#L48

Reply:
I think it does make sense to pass on the `value` to `SelectList` but maybe it should be called `selected` instead? I could image the SelectList by default highlights whatever item is selected, that's a pretty standard and useful UX feature you probably want in most (if not all) implementations.

Reply:
@G0L0L it's being passed to the `SelectListItem` which for the default `SelectList` is the `DefaultSelectListItem` which does not use it. @hongaar If you see us using it in the default item in the future though then we can keep it in

Reply:
@Stefanwullems @hongaar seems like Stefan is right. The `value` is not being used by `SelectList`. It's definitely a problem, since right now it's not possible to highlight selected value in the list.

I would keep `value` (actually renaming it to `selected` makes sense) prop for `Select` and modify `SelectList` to handle its `value` properly